templates/docs/examples/patterns/list-tree.html

```diff
@@ -56,14 +52,58 @@
   listTreeToggle.addEventListener('click', function(event) {
     event.preventDefault();
     var expand = this.getAttribute('aria-expanded') === 'true' ? false : true;
+
     toggleElement(listTreeToggle, expand);
   });
 }
 
+/**
+ * Adds or removes a class from an element.
+ * @param {HTMLElement} element - The element to add or remove the class from.
+ * @param {String} className - The class to add or remove.
+ * @param {Boolean} add - Whether to add or remove the class.
+ */
+function setClass(element, className, add) {
+  if (add) {
+    element.classList.add(className);
+  } else {
+    element.classList.remove(className);
+  }
+}
+
+/**
+ * Attaches event listeners for the list tree link.
+ * @param {HTMLElement} listTreeLink - `.p-list-tree__toggle` or `.p-list-tree__item` element to set up.
+ * @param {HTMLElement[]} otherTreeLinks - Other `.p-list-tree__toggle` or `.p-list-tree__item` elements on the page.
+ */
+function setupListTreeLink(listTreeLink, otherTreeLinks) {
```

Comment:
This behavior is intentional, to make the component behave a little bit more like a navigational component. I used [Carbon](https://react.carbondesignsystem.com/?path=/story/components-treeview--default) and [MUI](https://mui.com/x/react-tree-view/) as examples here.

As for the `is-active` styling vs using `aria-expanded`, I'm not sure if this can be achieved without `is-active`. This component was changed so that one of the nodes can be the "active" node (controlling some primary content to its right). The links themselves are not currently highlighted as active, only the grouping nodes (which have the `aria-expanded`).

Reply:
@pastelcyborg That's correct - my language was a bit imprecise, sorry. When I said "currently" I was referring to the current version of this component as released. 

In the released version, the groupings are highlighted grey when expanded, and there is no active styling for the nodes, just default link styling if the nodes are `a` elements.
<img width="232" alt="Screenshot 2024-08-29 at 3 10 47 PM" src="https://github.com/user-attachments/assets/c4a1dd76-b28a-4f81-9a79-926b2f166934">

In this proposed version, the groupings are never highlighted grey, and the nodes are colored as default text, unless they are active.

<img width="232" alt="Screenshot 2024-08-29 at 3 12 26 PM" src="https://github.com/user-attachments/assets/e40a692c-b6b1-41d3-8cf5-502ac179774d">